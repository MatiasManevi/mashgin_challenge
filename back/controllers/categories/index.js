const fs = require('fs');

const get = async (_, res) => {
	try {
		const categories = JSON.parse(fs.readFileSync('data/categories.json'))
		res.send(categories);
	} catch (e) {
		console.error(e);
		res.sendStatus(400);
	}
};

module.exports = {
	get
};
