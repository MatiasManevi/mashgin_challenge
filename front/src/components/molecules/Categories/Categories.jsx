import React from 'react';
import { useHistory } from 'react-router-dom';

import './Categories.scss';

const Categories = ({ categories }) => {
	const history = useHistory();

	return (
		<div className="categories">
			{categories.map((category, i) => {
				return (
					<div className="category__title" key={category.id}>
						<a
							onClick={() => {
								history.push(`?category=${category.id}`);
							}}
						>
							{category.name}
						</a>
					</div>
				);
			})}
		</div >
	);
};

export default Categories;
