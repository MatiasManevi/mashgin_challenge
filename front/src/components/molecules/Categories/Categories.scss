@import 'assets/scss/variables';

.categories {
	display: flex;
	padding: $space-32 0;
	justify-content: space-between;
	align-items: center;

	.category__title {
		font-size: $font-size-24;
		z-index: 1;
	}

	&:hover {
		cursor: pointer;
	}
}
