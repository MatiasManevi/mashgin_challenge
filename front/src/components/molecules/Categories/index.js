export { default } from 'components/molecules/Categories/Categories';
