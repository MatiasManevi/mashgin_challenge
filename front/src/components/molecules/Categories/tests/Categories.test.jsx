import React from 'react';
import { shallow } from 'enzyme';

import Categories from '../Categories';
import categories from './categories.fixtures';

describe('Categories', () => {
	it('should match snapshot', () => {
		const wrapper = shallow(<Categories categories={categories} />);
		expect(wrapper).toMatchSnapshot();
	});
});
