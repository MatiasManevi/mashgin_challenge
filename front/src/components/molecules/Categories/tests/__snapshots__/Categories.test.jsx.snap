// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories should match snapshot 1`] = `
<div
  className="categories"
>
  <div
    className="category__title"
    key="1"
  >
    <a
      onClick={[Function]}
    >
      Bakery
    </a>
  </div>
  <div
    className="category__title"
    key="2"
  >
    <a
      onClick={[Function]}
    >
      Entrees
    </a>
  </div>
  <div
    className="category__title"
    key="3"
  >
    <a
      onClick={[Function]}
    >
      Drinks
    </a>
  </div>
</div>
`;
